Automattic/pocket-casts-ios
# Fix: Update list views to use List

Fixes #1183

This PR refactors the uses of ScrollView/LazyStack in some screens to use Lists. 
This fixes layout bugs on WatchOS 10 and also brings the advantage of having reusable cells and also a nice animation effect when scrolling in new cells.


https://github.com/Automattic/pocket-casts-ios/assets/651601/047b0152-7d58-411d-8b6d-60a06769337a

| Before | After |
| - | - |
| ![Episodes_before](https://github.com/Automattic/pocket-casts-ios/assets/651601/664c3610-f056-471e-9abc-b4853459d764) |  ![Episodes_after](https://github.com/Automattic/pocket-casts-ios/assets/651601/3d0735b4-58b3-4f36-80ca-dfab196b4f4a) |
| ![NewReleases_before](https://github.com/Automattic/pocket-casts-ios/assets/651601/b92a6d31-116a-4cc0-847c-78251a18ae2a) | ![NewReleases_after](https://github.com/Automattic/pocket-casts-ios/assets/651601/f0e6b7fa-81b3-4770-bc2a-7fe9d76b0606) |
| ![InProgress_before](https://github.com/Automattic/pocket-casts-ios/assets/651601/7834c74b-1708-4166-92e3-bab84ce11119) | ![InProgress_after](https://github.com/Automattic/pocket-casts-ios/assets/651601/1b82ae7b-3212-4a77-a53d-722e6b39aaef) |
| ![Files_before](https://github.com/Automattic/pocket-casts-ios/assets/651601/a2e5462d-12ca-4855-b81f-0ee8ccaf4964) | ![Files_after](https://github.com/Automattic/pocket-casts-ios/assets/651601/0eec16ef-0805-4942-87e3-0735a10b3d71) |



## To test

 - Start the app on Watch OS version 10. If it's your first time using it on WatchOS you need to pair it with a simulator device that is already logged in and has some podcasts followed, some downloads and files to fully test this PR
 - Check the episodes list of a podcast and see if the layout match the screenshots above
 - Check the downloads list
 - Check the new releases list
 - Check the in progress list
 - Check the files list

The above should also be retested on WatchOS 9.0 to check if all works correctly there too.


## Checklist
